docs/spec.md

```diff
@@ -4963,8 +4969,8 @@ More formally:
 * (C9) `select` has type `(tensor<E>, tensor<E>) -> tensor<i1>` where
        `E = element_type(operand)`.
 * (C10) `scatter` has type `(tensor<E>, tensor<E>) -> tensor<E>` where
-        `E = element_type(operand)`.
-* (C11) `type(operand) = type(result)`.
+  `same_elementtype_ignoring_bitwidth(element_type(operand), element_type(E))`.
+* (C11) `baseline_type(operand) = baseline_type(result)`.
```

Comment:
```
same_elementtype_ignoring_bitwidth(element_type(E), element_type(result))
```

Reply:
Hi @loganchien 
When `operand`, `E` and `result` are all per-tensor quantized types,
`same_elementtype_ignoring_bitwidth` ensures that only the expressed types are the same, other components (storage_type, quantization params) may vary.
`baseline_element_type` ensures that both storage type and expressed type will be the same,  other components  (quantization params) may vary.

In this PR I am proposing `same_elementtype_ignoring_bitwidth(element_type(operand), element_type(E))` and `baseline_type(operand) = baseline_type(result)` which transitively means the expressed type of `E` and `result` to be the same and (storage type and quantization parameters) may vary, which will be the same constraint   with `same_elementtype_ignoring_bitwidth(element_type(E), element_type(result))`.

However, by using `baseline_type(operand) = baseline_type(result)`  we are adding an extra constraint that the storage type of `operand` and `result` will be the same, which is inline with the RFC proposal. 

Please let me know what you think about this

Reply:
>  by using baseline_type(operand) = baseline_type(result) we are adding an extra constraint that the storage type of operand and result will be the same

I don't think this extra constraint is desired nor inline with the RFC proposal.

One important use case for RFC is to read quantized type in ui8 storage type and then return a larger summation in i32.  This "extra" constraint defeat the purpose of the RFC.

Reply:
@loganchien 
updated the PR with the change.  

>> same_elementtype_ignoring_bitwidth(element_type(E), element_type(result))

Currently using
```
is_promotable(element_type(operand), element_type(result))
```

As `E` is not defined in the current constraint. However, transitively it would mean the same. Please let me know your opinion.